tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15684,12 +15718,11 @@ void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, llvm::StringRef StageName,
               << HLSLNodeObjectAttr::ConvertRecordTypeToStr(Kind);
       }
     }
-
+    unsigned int ArgIdx = 0;
```

Comment:
`ArgIdx` is basically used as a for loop variable.  I think the original purpose was to do something with this index, such as track whether the parameter referenced from `[MaxRecordsSharedWith(...)]` was the parameter with this attribute.  But as the comment says, this is checked elsewhere.  So, this `ArgIdx` is just used as a variable to look up each parameter in the function decl.

I would recommend the while loops using this be converted to for loops.  Since FunctionDecl has a `params()` method that gets the params, you can do away with `ArgIdx` entirely by using a different iteration pattern: `for (auto *ParamDecl : FD->params())`.